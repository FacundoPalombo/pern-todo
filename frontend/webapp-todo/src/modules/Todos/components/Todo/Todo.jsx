import React from "react";
import styles from "./Todo.module.scss";
import { Card } from "common";

export const Todo = ({ description = "" }) => {
  return (
    <Card>
      <p className={styles.paragraph}>{description}</p>
    </Card>
  );
};
