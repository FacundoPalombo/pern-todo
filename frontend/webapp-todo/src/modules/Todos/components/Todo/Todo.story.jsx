import React from "react";
import { Todo } from "./Todo";

export const withDescription = () => <Todo description="asdasldkÃ±a" />;
export const withoutDescription = () => <Todo />;

export default {
  title: "Todo",
  component: Todo,
};
