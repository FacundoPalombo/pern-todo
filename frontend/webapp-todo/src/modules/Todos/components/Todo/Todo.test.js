import React from "react";
import { Todo } from "./Todo";
import { render } from "@testing-library/react";

describe("GIVEN Todo component", () => {
  it("THEN should render", () => {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("THEN should render correctly when description is provided", () => {
    const { getByText } = render(<Todo description="Lol" />);
    const assert = getByText("Lol");
    expect(assert).not.toBeNull();
  });
});
