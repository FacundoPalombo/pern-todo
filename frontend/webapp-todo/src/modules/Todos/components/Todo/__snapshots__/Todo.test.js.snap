// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Todo component THEN should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p
      class="paragraph"
    />
  </div>
</DocumentFragment>
`;
